src/mmrelay/meshtastic/executors.py

```
@@ -0,0 +1,712 @@
+            stale_address,
+            max(timeout_count, reset_threshold),
+        )
+
+    with facade._ble_executor_lock:
+        if facade._ble_future and not facade._ble_future.done():
+            facade.logger.debug(
+                "BLE worker busy; skipping %s for %s", operation, ble_address
+            )
+            raise TimeoutError(
+                f"BLE {operation} already in progress for {ble_address}."
+            )
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)